### PR TITLE
Fix: load_error escaping invalid utf-8

### PR DESCRIPTION
**Pull request by dnamsons**
Closes https://github.com/artichoke/artichoke/issues/477

I hope that this is the proper way to go about solving the issue

### CODE REVIEW

File: artichoke-backend/src/extn/core/kernel/require.rs [hunk 1]
```diff
@@ -281,8 +281,8 @@ impl RelativePath {
...
 }
 
-fn load_error(interp: &Artichoke, filename: &[u8]) -> Result<LoadError, Exception> {
-    let mut message = String::from("cannot load such file -- ");
-    string::format_unicode_debug_into(&mut message, filename)?;
-    Ok(LoadError::new(interp, message))
+fn load_error(interp: &Artichoke, filename: &[u8]) -> LoadError {
+    let mut message = String::from("cannot load such file -- ").into_bytes();
+    message.extend_from_slice(filename);
+    LoadError::new_raw(&interp, message)
```
Comment (lopopolo) [inline artichoke-backend/src/extn/core/kernel/require.rs:287]: The extra `&` is not needed on the `interp`. `LoadError::new_raw` takes an `&Artichoke`, which `interp` already is. This extra reference gets us a `&&Artichoke` which is immediately dereferenced by the compiler to get us back the `&Artichoke` (this is called auto-deref).

There is no runtime impact, but can make the code trickier to read.

Comment (lopopolo) [reply artichoke-backend/src/extn/core/kernel/require.rs:287]:
```suggestion
    LoadError::new_raw(interp, message)
```

File: artichoke-backend/src/extn/core/kernel/require.rs [hunk 2]
```diff
@@ -281,8 +281,8 @@ impl RelativePath {
     }
 }
 
-fn load_error(interp: &Artichoke, filename: &[u8]) -> Result<LoadError, Exception> {
-    let mut message = String::from("cannot load such file -- ");
-    string::format_unicode_debug_into(&mut message, filename)?;
-    Ok(LoadError::new(interp, message))
+fn load_error(interp: &Artichoke, filename: &[u8]) -> LoadError {
+    let mut message = String::from("cannot load such file -- ").into_bytes();
```
Comment (lopopolo) [inline artichoke-backend/src/extn/core/kernel/require.rs:285]: As a matter of style, I dislike creating a `String` and "unparsing" into a `Vec` when we can create the `Vec` directly.

Comment (lopopolo) [reply artichoke-backend/src/extn/core/kernel/require.rs:285]:
```suggestion
    let mut message = b"cannot load such file -- ".to_vec();
```

**Comment by dnamsons** [comment]
Thank you for the suggestions on how to improve my code!

**Pull request merged**